### ISSUE TITLE
Add "share on social media" widgets

### ISSUE THREAD

**Issue by MarcSkovMadsen** [holoviz/panel#1115]
#### My Pain

I wanted to enable my users to easily share cool apps from the gallery on social media. 

A small detail is that I want users to be able to share a link directly to the location or state of my app. This is related to https://github.com/holoviz/panel/issues/811

Another detail is that the link should look like an icon from for example Font Awesome. And this has not yet been implemented in Panel. See https://github.com/holoviz/panel/pull/887

In general I think it would be nice for users to enable using sharing of their work with Panel on social media.

#### Solution

Add social media widgets (?) that enables users to easily create links to social media like

- facebook
- linkedin
- twitter
- reddit
- mail

#### Additional Context

At awesome-panel.org there is already this functionality.

![image](https://user-images.githubusercontent.com/42288570/75103638-b29de300-55fd-11ea-9dc9-c9bffc732c1e.png)

You can find the implementation here

https://github.com/MarcSkovMadsen/awesome-panel/blob/master/package/awesome_panel/express/fontawesome/share_link.py

the tests here

https://github.com/MarcSkovMadsen/awesome-panel/blob/master/package/tests/test_share_link.py

and an example app here 

https://github.com/MarcSkovMadsen/awesome-panel/blob/master/src/pages/gallery/awesome_panel_express_tests/test_share_links.py

A lot of the code would be transferable. I think the `view` function should be changed to a `_repr_html_` function. And then @philippjfr could guide on the implementation.

One thing though is that maybe it would be an idea to change the implementation from one based on html to one based on a button widget with an action to open the link. Lets call it a LinkButton. That would enable the app to record the action or do other stuff like displaying a "thanks for sharing" message for a few seconds.

**Comment by MarcSkovMadsen** [comment]
One thing for me to understand here is also if this is something to include in Panel? When I started awesome-panel.org I actually created an awesome-panel python package with an express module inspired by plotly.express that should enable me and others to even more quickly use Panel.

Is there some kind of overall vision on what to include and what not to include? Should Panel include functionality that is often used but can easily be created from primitives like Icons, Buttons, Markdown and Html if you know how?
